hermes/src/api.rs

```diff
@@ -26,15 +26,27 @@ mod rest;
 pub mod types;
 mod ws;
 
-#[derive(Clone)]
-pub struct ApiState {
-    pub state:     Arc<State>,
+pub struct ApiState<S = State> {
+    pub state:     Arc<S>,
     pub ws:        Arc<ws::WsState>,
     pub metrics:   Arc<metrics_middleware::Metrics>,
     pub update_tx: Sender<AggregationEvent>,
 }
 
-impl ApiState {
+/// Manually implement `Clone` as the derive macro will try and slap `Clone` on
+/// `State` which should not be Clone.
+impl<S> Clone for ApiState<S> {
+    fn clone(&self) -> Self {
+        Self {
+            state:     self.state.clone(),
+            ws:        self.ws.clone(),
+            metrics:   self.metrics.clone(),
+            update_tx: self.update_tx.clone(),
+        }
+    }
+}
```

Comment:
Hmmm it is strange that clone doesn't work as expected here. Can you explain more?

Reply:
Rust's Clone derive is a bit basic and will automatically add `T: Clone` for each generic. I _could_ mark `State` as `Clone` as it won't actually get cloned, but that introduces the risk a clone happens elsewhere in the future so opted for this.